### ISSUE TITLE
Amener le simulateur sur mobile

### ISSUE THREAD

**Issue by Thomas-Boutin** [datagir/nosgestesclimat-site#597]

**Comment by Thomas-Boutin** [comment]
Hello Mael!

Le souci d'encapsulation c'est que sur Android ça passera mais pas sur iOS, ils ne veulent pas d'encapsulation de ce genre sur l'AppStore :/

J'avais l'impression que réutiliser le moteur de règles n'était pas si coûteux.
Tu me diras si je me trompe, mais je crois que les règles de calcul sont toutes sur le repo. Cela ne serait pas faisable de les récupérer et de refaire le moteur sur mobile?

**Comment by laem** [comment]
Ah, merci intéressant comme retour. 

C'est en effet facile d'utiliser le moteur, mais la gestion du questionnaire est loin d'être simple, c'est un investissement de dev assez coûteux. Tant qu'il est possible de charger publicodes dans l'app, donc du Javascript, ça va car il n'y a "que" à reproduire l'interface qui utilise les résultats de publicodes. Mais ça reste substantiel. 

Après clairement, c'est possible, même dans une version simplifiée. Mais ça demande de recréer une base de code. Je t'avoue que de notre côté c'est assez peu envisageable :/

**Comment by Thomas-Boutin** [comment]
Ok je comprends.

Si nécessaire, pour la recréation de base de code, j'imagine que je pourrais le faire de mon côté. L'idée est que je fasse du dev ;-)
Est-ce qu'utiliser l'API REST décrite dans la doc est envisageable sinon? Si je comprends bien son fonctionnement, il faut parser les fichiers de règles et envoyer les "questions" à l'API?

**Comment by laem** [comment]
C'est une bonne question, je ne me suis pas encore trop intéressé à l'API reste de publi.codes. 

https://publi.codes/api-rest

J'imaginerais que l'API te renvoie les questions qu'il faut poser pour mener à bien l'évaluation de la variable centrale `bilan`. 

Désolé en tout cas de ne pas pouvoir te promettre d'aide pour une appli mobile, mais on a beaucoup d'autres priorités pour 2022 :)

**Comment by Thomas-Boutin** [comment]
Yes j'ai l'impression que ça marcherait en téléchargeant les fichiers de règles depuis GitHub. 
On pourrait imaginer de 

Pas de soucis pour l'aide, je comprends bien que vous ayez beaucoup de sujets :-D

Est-ce que tu penses que ça aurait du sens que je POC une appli mobile à partir des sources situées ici -> https://github.com/datagir/nosgestesclimat

**Comment by Thomas-Boutin** [comment]
Pour info nous avons continué les travaux, on va essayer d'avancer en RN sur https://github.com/Thomas-Boutin/nosgestesclimatmobile en reprenant le moteur et les règles qui sont partagées
